packages/react-scripts/template/README.md

```diff
@@ -1,68 +1,203 @@
-To learn React, check out the [React documentation](https://reactjs.org/).
+`server.js`:
 
-### Code Splitting
+```js
+const React = require('react');
+const express = require('express');
+const ReactDOMServer = require('react-dom/server');
 
-This section has moved here: https://facebook.github.io/create-react-app/docs/code-splitting
+const components = require('./my-app/build/ssr').default;
 
-### Analyzing the Bundle Size
+const router = express.Router();
 
-This section has moved here: https://facebook.github.io/create-react-app/docs/analyzing-the-bundle-size
+router.get('/', (req, res) => {
+  const element = React.createElement(components.App);
+  const html = ReactDOMServer.renderToString(element);
 
-### Making a Progressive Web App
+  res.render('index', { html });
+});
 
-This section has moved here: https://facebook.github.io/create-react-app/docs/making-a-progressive-web-app
+module.exports = router;
+```
 
-### Advanced Configuration
+`index.html`:
 
-This section has moved here: https://facebook.github.io/create-react-app/docs/advanced-configuration
+```html
+<div id="root">{{{html}}}</div>
+```
 
-### Deployment
+If you call `ReactDOM.render()` on a node that already has this server-rendered markup (`<div id="root">` in the example above), React will preserve it and only attach event handlers, allowing you to have a very performant first-load experience.
 
-This section has moved here: https://facebook.github.io/create-react-app/docs/deployment
+**Note:** Your external module imports (anything you import from `node_modules/` i.e. `import React from 'react';`) are bundled into the `ssr.js` output file. If you want to exclude any external modules from the output file for performance / code sharing reasons (i.e. if it's beign required elsewhere and is already in memory) you can do so by adding the following to your `package.json`:
 
-### `npm run build` fails to minify
+```json
+{
+  ...
+  "backpack-react-scripts": {
+    "ssrExternals": [
+      "react",
+      "react-dom"
+    ]
+  }
+}
+```
 
-This section has moved here: https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify
+## Compiling `node_modules` dependencies that contain JSX
+
+By default, modules imported from your app's `node_modules` directory will get compiled by Babel with the caveat that only standard ES features are supported. If you need to compile modules that contain non-standard ES features, including JSX, you can use the following:
+
+```json
+{
+  ...
+  "backpack-react-scripts": {
+    "babelIncludePrefixes": [
+      "my-module-prefix-",
+      "some-module"
+    ]
+  }
+}
+```
+
+The above example assumes that the module you want to compile is named with the prefix `my-module-prefix-` and a module with the name `some-module`. All entries in this array act as prefixes. It is used by all webpack configurations as well as the Jest configuration — if you want to avoid compiling a dependency in tests, you should mock it.
```

Comment:
```suggestion
The above example assumes that the module you want to compile is named with the prefix `my-module-prefix-` and a module with the name `some-module`. All entries in this array act as prefixes. It is used by all Webpack configurations as well as the Jest configuration — if you want to avoid compiling a dependency in tests, you should mock it.
```